### DISCUSSION TITLE
First Time build WLAN Pi on Raspberry Pi 4

### DISCUSSION THREAD

**Discussion by timg11** [WLAN-Pi/feedback#73] [General Inquiries]
I downloaded WLAN Pi OS, wrote it to an SD Card, and installed it in a Raspberry Pi 4 that I already had. 
I have an HDMI monitor, keyboard, and mouse. 
The Pi boots into the shell, and I can perform typical linux commands. 

I see that WLAN Pi ships with a small LCD display in the pre-made versions that provides a basic user interface. Is there an application that can give me similar UI functions but displaying on the HDMI monitor?

**Comment by joshschmelzle** [comment]
You will need to install a desktop environment. Just like the official Raspberry Pi OS Lite image, the WLAN Pi OS image does not ship with one.

**Comment by joshschmelzle** [comment]
Try installing the official Raspberry Pi one with `sudo apt install xserver-xorg raspberrypi-ui-mods` followed by `sudo raspi-config` and change the system boot option to boot to desktop. Let us know how it goes.

**Comment by timg11** [comment]
Thank you! I forgot that GUI could be added later. 
All steps completed - sudo apt install packages, enabled "desktop autologin" in raspi-config, let it reboot. The boot process on the console looked normal, but when it was time for the desktop to appear there was nothing but a black screen. 
Luckily I had enabled SSH while I was in raspi-config, so I was able to log in over SSH and change boot back to console. 

Also I'm noticing several minutes of delay during boot on "waiting for network to be configured", which eventually displays "Failed".
Finally, Ethernet gets an IP and works, but using raspi-config to configure WLAN fails.

**Reply by joshschmelzle** [reply]
Try `sudo systemctl disable systemd-networkd-wait-online.service` and rebooting to see if that fixes the boot time.

**Reply by timg11** [reply]
Thanks - that does fix the network hang. Any thoughts on the HDMI screen going black? I find that even if I change back to console and reboot, the screen remains black (I don't see the boot), until I power cycle.  Some setting for the video parameters in GUI mode needs changing?

**Reply by joshschmelzle** [reply]
I see the boot up via console whether using 1080p or 4K capture cards, but then the screen goes black as well.

**Reply by timg11** [reply]
> I see the boot up via console whether using 1080p or 4K capture cards, but then the screen goes black as well.

So  you never see the Pi desktop?  That is what I thought we were configuring when you said "change the system boot option to boot to desktop".  Instead of the Pi desktop, I get a permanent black screen until power cycles.

**Reply by joshschmelzle** [reply]
Correct, the desktop environment doesn't load and the HDMI output is just a black screen.

Either capture card I tried shows the boot process and start-up sequence scroll by prior to failing to load the desktop environment. I imagine yours is failing at the same spot. 

I haven't been able to figure out what is wrong yet. We don't officially support a desktop environment in our image because using the WLAN Pi via keyboard + mouse + monitor isn't a targeted use case, but at the same time, there should be a way to optionally enable that use case.

**Reply by timg11** [reply]
OK - I'll proceed with the terminal for now. Next step is to get the Netgear AXE3000 USB 3.0 WiFi Adapter -A8000 recognized and operating. I'll start a new thread for that.

**Reply by jiribrejcha** [reply]
@timg11 the Netgear adapter works out of the box.

**Reply by timg11** [reply]
@jiribrejcha  that's good to know. I'm sure I'm missing something obvious. My findings so far are summarize in [this post](https://github.com/orgs/WLAN-Pi/discussions/74)

I have searched for a while for a CLI guide on the wlanPi, but have not found it. Since I only have the terminal, I need to use the CLI for configuring wlanpi. I'm finding that "traditional" Pi utilities like iwconfig don't work as expected.

**Reply by joshschmelzle** [reply]
@timg11 `iwconfig` and wireless extensions is deprecated. You should be using `ip` and `iw` instead.
